src/owner.rs

```diff
@@ -0,0 +1,357 @@
+        // check that `team` is the `slug` in results, and grab its `id`
+
+        // "sanitization"
+        fn whitelist(c: &char) -> bool {
+            match *c {
+                'a'...'z' | 'A'...'Z' | '0'...'9' | '-' | '_' => false,
+                _ => true
+            }
+        }
+
+        if let Some(c) = org_name.chars().find(whitelist) {
+            return Err(human(format!("organization cannot contain special characters like {}",
+                                   c)));
+        }
+
+        let resp = try!(http::handle()
+                         .get(format!("https://api.github.com/orgs/{}/teams", org_name))
+                         .header("Accept", "application/vnd.github.v3+json")
+                         .header("User-Agent", "hello!")
+                         .header("Authorization", &format!("token {}", &req_user.gh_access_token))
+                         .exec());
+
+        match resp.get_code() {
+            200 => {} // Ok!
+            403 => {
+                return Err(human("It looks like you don't have permission \
+                                  to query one of these organizations. \
+                                  You may need to re-authenticate on \
+                                  crates.io to grant permission to read \
+                                  github org memberships. Just go to \
+                                  https://crates.io/login"));
+            }
+            _ => {
+                return Err(internal(format!("didn't get a 200 result from github: {}",
+                                            resp)));
+            }
+        }
+
+        #[derive(RustcDecodable)]
+        struct GithubTeam {
+            slug: String,
+            id: i32,
+        }
+
+        let json = try!(str::from_utf8(resp.get_body()).ok().chain_error(||{
+            internal("github didn't send a utf8-response")
+        }));
+        let teams: Vec<GithubTeam> = try!(json::decode(json).chain_error(|| {
+            internal("github didn't send a valid json response")
+        }));
+
+        let mut github_id = None;
+
+        for team in teams {
+            if team.slug == team_name {
+                github_id = Some(team.id);
+                break;
+            }
+        }
+
+        let github_id = try!(github_id.ok_or_else(|| {
+            human(format!("could not find the github team {}/{}", org_name, team_name))
+        }));
+
+        // mock Team (only need ID to check team status)
+        let team = Team { github_id: github_id, cargo_id: 0, name: String::new() };
+        if !try!(team.contains_user(req_user)) {
+            return Err(human("only members of a team can add it as an owner"));
+        }
+
+        // insert into DB for reals
+        try!(conn.execute("INSERT INTO teams
+                           (name, github_id)
+                           VALUES ($1, $2)",
+                          &[&name, &github_id]));
+
+        // read it right back out:
```

Comment:
If you search for "RETURNING" you can see a method of inserting while also returning the row that was just inserted